neqo-common/src/datagram.rs

```diff
@@ -117,10 +133,115 @@ impl<D: AsRef<[u8]>> AsRef<[u8]> for Datagram<D> {
     }
 }
 
+/// A batch of [`Datagram`]s with same metadata, e.g. destination.
+///
+/// Upholds Linux GSO requirement. That is, all but the last datagram in the
+/// batch have the same size. The last datagram may be equal or smaller.
+#[derive(Clone, PartialEq, Eq)]
+pub struct DatagramBatch {
+    src: SocketAddr,
+    dst: SocketAddr,
+    tos: IpTos,
+    segment_size: usize,
+    d: Vec<u8>,
+}
```

Comment:
Following @martinthomson suggestion in https://github.com/mozilla/neqo/pull/2093#discussion_r1785760615, though calling it `DatagramBatch` instead of `DatagramVec` or `DatagramList`. Let me know if you feel strongly about any of these names.